### PR TITLE
[DPE-3724] Move configuration hub to use a rock image

### PR DESCRIPTION
**Pull request by welpaolo**

### CODE REVIEW

File: src/core/workload.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        """Checks that the workload is active."""
+        ...
+
+    @abstractmethod
+    def ready(self) -> bool:
+        """Checks that the container/snap is ready."""
+        ...
+
+    @staticmethod
+    def generate_password() -> str:
```
Comment (deusebio) [inline src/core/workload.py]: **question** do we use this? if not, maybe we can remove it

File: src/managers/k8s.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+"""Kubernetes manager."""
+
+from lightkube.core.client import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.resources.core_v1 import ServiceAccount
+
+from common.utils import WithLogging
+
+
+class KubernetesManager(WithLogging):
```
Comment (deusebio) [inline src/managers/k8s.py:14]: **praise** I like to abstract away all K8s operations into a proper manager. This could indeed be used in multiple charms.

**Review by deusebio — APPROVED** [review]
Super nice!!! It is also very nice that we are starting to have components, that are consistent and that could be shared between different charms (e.g. S3, K8s).

Very well done! I could follow and also review this code very quickly, as most of the code was somewhat familiar!!

File: src/core/context.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+from enum import Enum
+
+from charms.data_platform_libs.v0.data_interfaces import RequirerData
+from ops import ActiveStatus, BlockedStatus, CharmBase, MaintenanceStatus, Relation
+
+from common.utils import WithLogging
+from core.domain import S3ConnectionInfo
+
+S3 = "s3-credentials"
```
Comment (theoctober19th) [inline src/core/context.py]: nit: could this live probably in `constants.py` as well?

**Review by theoctober19th — APPROVED** [review]
Looks good to me.

File: src/core/workload.py (hunk 1 above)
Comment (welpaolo) [reply src/core/workload.py]: sure!

File: src/core/context.py (hunk 3 above)
Comment (welpaolo) [reply src/core/context.py]: Done

**Pull request merged**